Sources/SwiftParser/Declarations.swift

```diff
@@ -1387,7 +1392,7 @@ extension Parser {
 
         let type: RawTypeSyntax?
         if shouldParseType {
-          type = self.parseType()
+          type = self.parseType(misplacedSpecifiers: misplacedSpecifiers)
```

Comment:
This name is somewhat confusing since function types can also have specifiers, but this is for the type specifiers specifically.